components/builder-worker/src/runner/util.rs

```diff
@@ -0,0 +1,222 @@
+                prj_integration.get_integration_name()
+            )));
+        }
+        // TODO fn: use a struct and serde to do heavy lifting
+        let opts: JsonValue = match serde_json::from_str(prj_integration.get_body()) {
+            Ok(json) => json,
+            Err(err) => {
+                return Err(Error::InvalidIntegrations(format!(
+                    "project integration body does not deserialize as JSON: {:?}",
+                    err
+                )))
+            }
+        };
+        // Required keys with string values
+        for str_key in vec!["docker_hub_repo_name"].iter() {
+            match opts.get(str_key) {
+                Some(val) => {
+                    if !val.is_string() {
+                        return Err(Error::InvalidIntegrations(format!(
+                            "project integration {} value must be a string",
+                            str_key
+                        )));
+                    }
+                }
+                None => {
+                    return Err(Error::InvalidIntegrations(
+                        format!("project integration {} missing", str_key),
+                    ));
+                }
+            }
+        }
+        // Required keys with boolean values
+        for bool_key in vec!["latest_tag", "version_tag", "version_release_tag"].iter() {
+            match opts.get(bool_key) {
+                Some(val) => {
+                    if !val.is_boolean() {
+                        return Err(Error::InvalidIntegrations(format!(
+                            "project integration {} value must be a bool",
+                            bool_key
+                        )));
+                    }
+                }
+                None => {
+                    return Err(Error::InvalidIntegrations(
+                        format!("project integration {} missing", bool_key),
+                    ));
+                }
+            }
+        }
+        // Optional keys with string values
+        if let Some(val) = opts.get("custom_tag") {
+            if !val.is_string() {
+                return Err(Error::InvalidIntegrations(format!(
+                    "project integration custom_tag value must be a string"
+                )));
+            }
+        }
+    }
+    // Validate origin integration
+    {
+        let org_integrations = workspace.job.get_integrations();
+        if org_integrations.is_empty() {
+            return Err(Error::InvalidIntegrations(format!(
+                "missing Docker credentials from origin integrations"
+            )));
+        }
+        let org_integration = org_integrations.first().unwrap();
+        if org_integration.get_integration() != "docker" {
+            return Err(Error::InvalidIntegrations(format!(
+                "origin integration '{}' not supported",
+                org_integration.get_integration()
+            )));
+        }
+        if org_integration.get_name() != "docker" {
```

Comment:
I think the original intent was to have the creds name be the username, but if the UI is hardcoding this to docker, then this is ok for now.